hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCleanHelper.java

```diff
@@ -16,6 +16,9 @@
 
```

Comment:
In general, the logic of pending compactions file slices removal is very interleaved with the cleaner code, can you separate them out ? Something like getAllEligibleFileSlices() where one function removed all the file slices from the list that should not be compacted and the other just does what this code used to do ? There is an extra loop over the file slices but I feel that will be ok given clean is a one time operation for some batches anyways. Thoughts ?

Reply:
Lets revisit further refactoring holistically after we make a push on finishing up pending features related to async compaction and usability.

Reply:
I'm okay with that.